src/Components/Web.JS/src/Rendering/JSRootComponents.ts

```diff
@@ -10,80 +11,107 @@ let manager: DotNet.DotNetObject | undefined;
 
 // These are the public APIs at Blazor.rootComponents.*
 export const RootComponentsFunctions = {
-    async add(toElement: Element, componentIdentifier: string, initialParameters: ComponentParameters): Promise<DynamicRootComponent> {
-        if (!initialParameters) {
-            throw new Error('initialParameters must be an object, even if empty.');
-        }
-
-        // Track the container so we can use it when the component gets attached to the document via a selector
-        const containerIdentifier = pendingRootComponentContainerNamePrefix + (++nextPendingDynamicRootComponentIdentifier).toString();
-        pendingRootComponentContainers.set(containerIdentifier, toElement);
-
-        // Instruct .NET to add and render the new root component
-        const componentId = await getRequiredManager().invokeMethodAsync<number>(
-            'AddRootComponent', componentIdentifier, containerIdentifier);
-        const component = new DynamicRootComponent(componentId);
-        await component.setParameters(initialParameters);
-        return component;
+  async add(toElement: Element, componentIdentifier: string, initialParameters: ComponentParameters): Promise<DynamicRootComponent> {
```

Comment:
I apologize for the weird diffs in this file - I noticed that this file used four-space indents while the ESLint configuration requires two-space indents, so a lot of these diffs are just from changing the indentation.

The relevant changes are in the `DynamicRootComponent` class.